airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt

```diff
@@ -295,19 +297,32 @@ open class S3StorageOperations(
         cleanUpBucketObject(objectPath, listOf())
     }
 
-    private fun listObjects(objectPath: String): ObjectListing {
+    private fun listObjectsByPage(
+        objectPath: String,
+        maxKeysPerPage: Int = 1000,
+        pageConsumer: (List<S3ObjectSummary>) -> Unit
+    ) {
```

Comment:
are you intentionally keeping the regex matching stuff outside this method? (i.e. the `Pattern.compile(getRegexFormat(namespace, streamName, pathFormat))` + `objectSumaries.filter { regex.matcher(it).matches() }` thing)

Reply:
yeah. wasn't sure if all usages are doing filter at the time of refactoring (turns out they seem to be)